apps/mobile/src/utils/calendar.ts

```diff
@@ -0,0 +1,276 @@
+        )
+
+        if (!calendar) {
+          const calendarId = await Calendar.createCalendarAsync(vexlCalendar)
+          set(vexlCalendarIdAtom, calendarId)
+
+          return E.right(calendarId)
+        }
+
+        return E.right(calendar.id)
+      } catch (error) {
+        return E.left({_tag: 'calendarError', reason: 'UnknownError', error})
+      }
+    }
+  })
+
+function createCalendarEvent({
+  calendarEventId,
+  calendarId,
+  event,
+}: {
+  calendarEventId: CalendarEventId | undefined
+  calendarId: string
+  event: Omit<Partial<Event>, 'id'>
+}): TE.TaskEither<CalendarError, CalendarEventId> {
+  return async () => {
+    try {
+      if (calendarEventId) {
+        const existingEvent = await Calendar.getEventAsync(
+          calendarEventId
+        ).catch(() => undefined)
+        if (existingEvent) {
+          return E.left({_tag: 'calendarError', reason: 'EventAlreadyExists'})
+        }
+      }
+
+      const eventId = await Calendar.createEventAsync(calendarId, event)
+      return E.right(eventId as CalendarEventId)
+    } catch (error) {
+      return E.left({_tag: 'calendarError', reason: 'UnknownError', error})
+    }
+  }
+}
+
+export const addEventToCalendarActionAtom = atom(
+  null,
+  async (
+    get,
+    set,
+    {
+      calendarEventIdAtom,
+      event,
+    }: {
+      calendarEventIdAtom: Atom<CalendarEventId | undefined>
+      event: Omit<Partial<Event>, 'id'>
+    }
+  ): Promise<CalendarEventId | undefined> => {
+    const {t} = get(translationAtom)
+    const calendarEventId = get(calendarEventIdAtom)
+
+    set(loadingOverlayDisplayedAtom, true)
+
+    return await pipe(
+      TE.Do,
+      TE.bindW('calendarId', () =>
+        set(
+          createCalendarIfNotExistsAndTryToResolvePermissionsAlongTheWayActionAtom
+        )
+      ),
+      TE.bindW('calendarEventId', ({calendarId}) =>
+        createCalendarEvent({
+          calendarEventId,
+          calendarId,
+          event,
+        })
+      ),
+      TE.chainFirstW(() =>
+        set(askAreYouSureActionAtom, {
+          steps: [
+            {
+              type: 'StepWithText',
+              title: t('tradeChecklist.eventAddedSuccess'),
+              description: t('tradeChecklist.eventAddedSuccessDescription'),
+              positiveButtonText: t('common.close'),
+            },
+          ],
+          variant: 'info',
+        })
+      ),
+      TE.match(
+        (e) => {
+          set(loadingOverlayDisplayedAtom, false)
+          if (e._tag === 'calendarError') {
+            if (e.reason === 'UserCalendarNotFound') {
+              Alert.alert(t('tradeChecklist.noUserCalendarFound'))
+            }
+            if (e.reason === 'EventAlreadyExists') {
+              Alert.alert(t('tradeChecklist.eventAlreadyExists'))
+            }
+
+            if (e.reason === 'UnknownError') {
+              reportError('error', new Error('Error creating calendar event'), {
```

Comment:
What about all the other reasons. If they are not displayed to users is it ok to omit reporting them?